### PR TITLE
Improve performance of get account by ID on Citus

### LINKED ISSUE #5228: Citus optimize get account by id query
### Problem

REST API performance test with citus db shows get account by account ID REST API performs poorly.

### Solution

- Run `explain analyze` to understand the bottleneck of the SQL query, and determine if the performance issue is from the account info query or the account related transaction query
- Try not to use the shared `getAccountQuery` and write and optimize dedicated get account by account id query
- Optionally, research ways to optimize get account related transactions query
- Run performance test to verify

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by jnels124**
**Description**:

Modify get account by id query to filter on single account id

**Related issue(s)**:

Fixes #5228

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/accounts.js [hunk 1]
```diff
@@ -299,9 +295,9 @@ const getOneAccount = async (req, res) => {
   const resultTypeQuery = utils.parseResultParams(req);
   const {query, params, order, limit} = utils.parseLimitAndOrderParams(req);
 
-  const accountIdParams = [encodedId];
+  const accountIdParams = [encodedId, encodedId];
   const {query: entityQuery, params: entityParams} = getAccountQuery(
-    {query: 'e.id = ?', params: accountIdParams},
+    {query: 'e.id = ? and (es.id = ? OR es.id IS NULL)', params: accountIdParams},
```
Comment (steven-sheehy) [inline hedera-mirror-rest/accounts.js:300]: Why would this provide any benefit? The `left join entity_stake es on es.id = e.id` should already optimally join.

Comment (steven-sheehy) [inline hedera-mirror-rest/accounts.js:300]: Need k7 report.md results from before and after. Also need to ensure no regression on v1.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-rest/accounts.js (hunk 1 above)
Comment (jnels124) [reply hedera-mirror-rest/accounts.js:300]: agreed but it is enough to hint the citus planner to only send to the correct node.

Comment (jnels124) [reply hedera-mirror-rest/accounts.js:300]: I ensured only minor regression on v1 (-20 ms average difference, -50 RPS). I ran for v2 and there was a decrease of about 4s. I will drop some screen shots here

Comment (jnels124) [reply hedera-mirror-rest/accounts.js:300]: with-changes
without changes

Comment (steven-sheehy) [reply hedera-mirror-rest/accounts.js:300]: Not sure what improvement you're seeing. Looks like 86 rps in both. But again, that output is not accurate. We need the `report.md` output and in particular the pass RPS.

Comment (jnels124) [reply hedera-mirror-rest/accounts.js:300]: When I ran things for the first pass, the before was 21.2 s average and the after was 17.3 s average. The second pass, is what these screenshots are and yes they are pretty much identical. That was actually to be expected though. This runs in parallel with the get account related transactions query which is less performant than the entity query.

Comment (steven-sheehy) [reply hedera-mirror-rest/accounts.js:300]: Perhaps manually disable transactions query and re-run single test. But I want the output of report.md, not k6 stdout.

Comment (jnels124) [reply hedera-mirror-rest/accounts.js:300]: That is a useful test to run. However, report.md can't be generated when running a single test.

Comment (jnels124) [reply hedera-mirror-rest/accounts.js:302]: ## V1 Before
| Scenario | URL | VUS | Pass% | RPS | Pass RPS | Avg. Req Duration | Comment |
|----------|----------|-----|-------|-----|----------|-------------------|---------|
| accounts | /accounts | 1500 | 100.00 | 605.29/s | 605.29/s | 1062.93ms | |
| accountsBalanceFalse | /accounts?balance=false | 1500 | 100.00 | 1708.71/s | 1708.71/s | 431.99ms | |
| accountsBalanceFalsePubkey | /accounts?balance=false&account.publickey={publicKey} | 1500 | 100.00 | 5720.93/s | 5720.93/s | 167.64ms | |
| accountsBalanceGt0 | /accounts?account.balance=gt:0 | 1500 | 100.00 | 1690.43/s | 1690.43/s | 425.52ms | |
| accountsBalanceGt0Pubkey | /accounts?account.balance=gt:0&account.publickey={publicKey} | 1500 | 100.00 | 5461.20/s | 5461.20/s | 190.01ms | |
| accountsBalanceNe | /accounts?account.balance=ne:{balance}&order=desc | 1500 | 99.99 | 2154.36/s | 2154.14/s | 252.22ms | |
| accountsId | /accounts/{accountId} | 1500 | 100.00 | 5615.04/s | 5615.04/s | 155.31ms | |
| accountsIdNe | /accounts?account.id=ne:{accountId}&order=desc | 1500 | 100.00 | 2165.72/s | 2165.72/s | 250.63ms | |

## V1 After
| Scenario | URL | VUS | Pass% | RPS | Pass RPS | Avg. Req Duration | Comment |
|----------|----------|-----|-------|-----|----------|-------------------|---------|
| accounts | /accounts | 1500 | 100.00 | 601.91/s | 601.91/s | 1026.58ms | |
| accountsBalanceFalse | /accounts?balance=false | 1500 | 100.00 | 1693.33/s | 1693.33/s | 430.24ms | |
| accountsBalanceFalsePubkey | /accounts?balance=false&account.publickey={publicKey} | 1500 | 100.00 | 5525.86/s | 5525.86/s | 165.50ms | |
| accountsBalanceGt0 | /accounts?account.balance=gt:0 | 1500 | 100.00 | 1693.18/s | 1693.18/s | 430.18ms | |
| accountsBalanceGt0Pubkey | /accounts?account.balance=gt:0&account.publickey={publicKey} | 1500 | 100.00 | 5619.96/s | 5619.96/s | 191.93ms | |
| accountsBalanceNe | /accounts?account.balance=ne:{balance}&order=desc | 1500 | 100.00 | 2183.65/s | 2183.65/s | 250.21ms | |
| accountsId | /accounts/{accountId} | 1500 | 100.00 | 5649.88/s | 5649.88/s | 154.13ms | |
| accountsIdNe | /accounts?account.id=ne:{accountId}&order=desc | 1500 | 100.00 | 2176.97/s | 2176.97/s | 251.57ms | |

## V2 Before
| Scenario | URL | VUS | Pass% | RPS | Pass RPS | Avg. Req Duration | Comment |
|----------|----------|-----|-------|-----|----------|-------------------|---------|
| accounts | /accounts | 1500 | 69.35 | 87.45/s | 60.65/s | 15756.76ms | |
| accountsBalanceFalse | /accounts?balance=false | 1500 | 96.09 | 131.17/s | 126.04/s | 11021.34ms | |
| accountsBalanceFalsePubkey | /accounts?balance=false&account.publickey={publicKey} | 1500 | 98.40 | 145.35/s | 143.02/s | 10125.77ms | |
| accountsBalanceGt0 | /accounts?account.balance=gt:0 | 1500 | 86.25 | 101.66/s | 87.68/s | 13828.54ms | |
| accountsBalanceGt0Pubkey | /accounts?account.balance=gt:0&account.publickey={publicKey} | 1500 | 93.96 | 118.75/s | 111.58/s | 12097.88ms | |
| accountsBalanceNe | /accounts?account.balance=ne:{balance}&order=desc | 1500 | 89.74 | 106.89/s | 95.92/s | 13280.40ms | |
| accountsId | /accounts/{accountId} | 1500 | 99.94 | 231.74/s | 231.60/s | 6449.45ms | |
| accountsIdNe | /accounts?account.id=ne:{accountId}&order=desc | 1500 | 89.28 | 107.84/s | 96.28/s | 13100.56ms | |

## V2 After
| Scenario | URL | VUS | Pass% | RPS | Pass RPS | Avg. Req Duration | Comment |
|----------|----------|-----|-------|-----|----------|-------------------|---------|
| accounts | /accounts | 1500 | 69.79 | 87.03/s | 60.74/s | 15901.94ms | |
| accountsBalanceFalse | /accounts?balance=false | 1500 | 96.89 | 134.53/s | 130.35/s | 10820.26ms | |
| accountsBalanceFalsePubkey | /accounts?balance=false&account.publickey={publicKey} | 1500 | 98.59 | 147.85/s | 145.77/s | 9966.11ms | |
| accountsBalanceGt0 | /accounts?account.balance=gt:0 | 1500 | 86.71 | 102.03/s | 88.47/s | 13801.66ms | |
| accountsBalanceGt0Pubkey | /accounts?account.balance=gt:0&account.publickey={publicKey} | 1500 | 94.00 | 120.14/s | 112.93/s | 11943.26ms | |
| accountsBalanceNe | /accounts?account.balance=ne:{balance}&order=desc | 1500 | 89.61 | 108.89/s | 97.58/s | 12978.47ms | |
| accountsId | /accounts/{accountId} | 1500 | 100.00 | 2784.57/s | 2784.57/s | 501.56ms | |
| accountsIdNe | /accounts?account.id=ne:{accountId}&order=desc | 1500 | 89.82 | 107.65/s | 96.69/s | 13152.19ms | |

Comment (jnels124) [reply hedera-mirror-rest/accounts.js:302]: need to pass the limit. here will do that

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

**Review by edwin-greene — APPROVED** [review]
Looks good!

**Pull request merged**